app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -8,34 +8,46 @@ import com.lightricks.feedexercise.database.UserProject
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
 import org.junit.Assert
-import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
-class FeedRepositoryTest(private var db : FeedDatabase,
-                         private var feedRepository: FeedRepository,
-                         private var userProjectList : List<UserProject>) {
-
-    @Before
-    fun init(){
-        this.db = Room.inMemoryDatabaseBuilder(ApplicationProvider.getApplicationContext(),
-            FeedDatabase::class.java).build()
-        this.feedRepository = FeedRepository(MockApiService,db)
+class FeedRepositoryTest() {
+    private val db: FeedDatabase = Room.inMemoryDatabaseBuilder(
+        ApplicationProvider.getApplicationContext(),
+        FeedDatabase::class.java
+    ).build()
+    private val feedRepository: FeedRepository = FeedRepository(ConstantItemsApiService, db)
+    private val userProjectList: List<UserProject> = Constans.hardCodedList.map { item ->
+        UserProject(item.id, responseUrlToThumbnailUrl(item.thumbnailUrl), item.isPremium)
     }
 
+
     @Test
-    fun testRefreshSaves(){
+    fun testRefreshSaves() {
         feedRepository.refresh().test().awaitTerminalEvent()
-        var fetchedList : List<UserProject>
+        var fetchedList: List<UserProject>
         db.userProjectDao().getAll().subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
-            .doOnNext{userProjects ->
+            .doOnNext { userProjects ->
                 fetchedList = userProjects
-                Assert.assertEquals(fetchedList,userProjectList)
+                Assert.assertEquals(fetchedList, userProjectList)
             }
             .subscribe()
     }
+
+    @Test
+    fun testGetAll(){
+        feedRepository.refresh().test().awaitTerminalEvent()
```

Comment:
This test is not supposed to call `refresh()`.
1. Insert the items directly into the DB
2. Call `feedRepository.getAllProjects()`
3. Assert you got the expected items